common/db-scripts/db-DataGeneration/inventory_creation.sql

```diff
@@ -0,0 +1,382 @@
+-- FUNCTION: public.generate_inventory_data(integer)
+
+-- DROP FUNCTION IF EXISTS public.generate_inventory_data(integer);
+
+CREATE OR REPLACE FUNCTION public.generate_inventory_data(
+	num_entries integer)
+    RETURNS void
+    LANGUAGE 'plpgsql'
+    COST 100
+    VOLATILE PARALLEL UNSAFE
+AS $BODY$
+DECLARE
+  instance_id UUID;
+  srs_id UUID;
+  instance_hrid TEXT;
+  instance_title TEXT;
+  holding_id UUID;
+  holding_hrid TEXT;
+  item_id UUID;
+  item_barcode TEXT;
+  item_hrid TEXT;
+  creation_date TIMESTAMP;
+  created_by_user UUID;
+  instancestatusid UUID := NULL;
+  instance_type_id UUID;
+  mode_of_issuance_id UUID;
+  holding_type_id UUID;
+  callnumber_type_id UUID;
+  material_type_id UUID;
+  permanent_loan_type_id UUID;
+  permanent_location_id UUID;
+  complete_updated_date TIMESTAMP;
+  snapshot_id UUID;
+  sourceid UUID;
+  itemNoteTypeId UUID;
+  conflict_found BOOLEAN;
+BEGIN
+  FOR i IN 1..num_entries LOOP
+    -- Generate new UUIDs
+    instance_id := public.uuid_generate_v4();
+    holding_id := public.uuid_generate_v4();
+    item_id := public.uuid_generate_v4();
+    srs_id := public.uuid_generate_v4();
+    --getting user with admin parmissuion
+     SELECT jsonb->>'userId' INTO created_by_user
+    FROM [tenant]_mod_permissions.permissions_users
+    WHERE jsonb -> 'permissions' @> '"folio_admin"' 
+    LIMIT 1;
+--getting instance_type_id
+    SELECT id into instance_type_id
+	  FROM [tenant]_mod_inventory_storage.instance_type limit 1;
+--getting mode_of_issuance_id
+    SELECT id into mode_of_issuance_id
+	  FROM [tenant]_mod_inventory_storage.mode_of_issuance limit 1;
+--getting holding_type_id
+    SELECT id into holding_type_id
+	  FROM [tenant]_mod_inventory_storage.holdings_type limit 1;
+--getting callnumber_type_id
+     SELECT id into callnumber_type_id
+	   FROM [tenant]_mod_inventory_storage.call_number_type limit 1;
+--getting material_type_id
+      SELECT id into material_type_id
+    	FROM [tenant]_mod_inventory_storage.material_type limit 1;
+--getting permanent_loan_type_id
+      SELECT id into permanent_loan_type_id
+	    FROM [tenant]_mod_inventory_storage.loan_type limit 1;
+--getting permanent_location_id
+      SELECT id into permanent_location_id
+     	FROM [tenant]_mod_inventory_storage.location limit 1;
+
+--getting snapshot_id
+      SELECT id into snapshot_id
+      	FROM [tenant]_mod_source_record_storage.snapshots_lb  limit 1;
+        
+--getting sourceid
+      SELECT id into sourceid
+      	FROM  [tenant]_mod_inventory_storage.holdings_records_source 
+        where jsonb->>'name' ='FOLIO'  limit 1;
+
+--getting itemNoteTypeId
+      SELECT id into itemNoteTypeId
+      	FROM [tenant]_mod_inventory_storage.instance_note_type limit 1;
+
+
+    -- Generate unique HRIDs
+    LOOP
```

Comment:
Perhaps these loops are what makes the script runs long. Could we have a separate script that generates these unique entities, store them in temp tables which can be queried at runtime of this script? It's not going to make it any faster, but at least we separate out the two processings so that we could run either part at the appropriate time rather than having to run all at once?